src/test/java/core/basesyntax/service/impl/DataParserServiceImplTest.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class DataParserServiceImplTest {
+    private DataParserServiceImpl dataParserService;
+
+    @BeforeEach
+    void setUp() {
+        dataParserService = new DataParserServiceImpl();
+    }
+
+    @DisplayName("Checking for passing null as a lines value")
+    @Test
+    void parseDate_nullLinesValue_notOk() {
+        assertThrows(NullPointerException.class, () -> dataParserService.parseData(null));
+    }
+
+    @DisplayName("Checking for passing empty list")
+    @Test
+    void parseData_emptyLinesList_ok() {
+        List<FruitTransaction> expected = Collections.emptyList();
+        List<FruitTransaction> actual = dataParserService.parseData(Collections.emptyList());
+        assertEquals(expected, actual);
```

Comment:
Let's better use the next approach for test empty lists:
```suggestion
        assertTrue(actual.isEmpty());
```
Do not forget remove redundant variables.